internal/pkg/docker/dockercompose/svc_image.go

```diff
@@ -0,0 +1,144 @@
+		return manifest.Image{}, nil, errors.New("missing one of `build` or `image`")
+	}
+
+	ignored, err := unsupportedBuildKeys(build)
+	if err != nil {
+		return manifest.Image{}, ignored, err
+	}
+
+	args, err := convertMappingWithEquals(build.Args)
+	if err != nil {
+		return manifest.Image{}, nil, fmt.Errorf("convert build args: %w", err)
+	}
+
+	buildArgs := manifest.DockerBuildArgs{
+		Context:    nilIfEmpty(build.Context),
+		Dockerfile: nilIfEmpty(build.Dockerfile),
+		Args:       args,
+		Target:     nilIfEmpty(build.Target),
+		CacheFrom:  build.CacheFrom,
+	}
+
+	image.Build = manifest.BuildArgsOrString{
+		BuildArgs: buildArgs,
+	}
+
+	return image, ignored, nil
+}
+
+func nilIfEmpty(s string) *string {
+	if s == "" {
+		return nil
+	} else {
+		return &s
+	}
+}
+
+// convertMappingWithEquals checks for entries with missing values and generates an error if any are found.
+func convertMappingWithEquals(inArgs types.MappingWithEquals) (map[string]string, error) {
+	args := map[string]string{}
+	var badArgs []string
+
+	for k, v := range inArgs {
+		if v == nil {
+			badArgs = append(badArgs, k)
+		} else {
+			args[k] = *v
+		}
+	}
+
+	if len(badArgs) != 0 {
+		return nil, fmt.Errorf("%s '%v' %s missing %s and %s user input, this is unsupported in Copilot",
+			english.PluralWord(len(badArgs), "entry", "entries"), badArgs,
+			english.PluralWord(len(badArgs), "is", "are"),
+			english.PluralWord(len(badArgs), "a value", "values"),
+			english.PluralWord(len(badArgs), "requires", "require"))
+	}
+
+	if len(args) == 0 {
+		return nil, nil
+	}
+
+	return args, nil
+}
+
+// unsupportedBuildKeys checks for unsupported keys in the given Compose build config.
+func unsupportedBuildKeys(build *types.BuildConfig) (IgnoredKeys, error) {
+	if build.SSH != nil || build.Secrets != nil {
+		return nil, errors.New("`build.ssh` and `build.secrets` are not supported yet, see " +
+			"https://github.com/aws/copilot-cli/issues/2090 for details")
+	}
+
+	if build.ExtraHosts != nil {
+		return nil, errors.New("key `build.extra_hosts` is not supported yet, this might break your app")
+	}
+
+	if build.Network != "" {
+		return nil, errors.New("key `build.network` is not supported yet, this might break your app")
+	}
+
+	var ignoredKeys []string
+
+	if build.CacheTo != nil {
+		ignoredKeys = append(ignoredKeys, "build.cache_to")
+	}
+
+	if build.NoCache {
+		ignoredKeys = append(ignoredKeys, "build.no_cache")
+	}
+
+	if build.Pull {
+		ignoredKeys = append(ignoredKeys, "build.pull")
+	}
+
+	if build.Isolation != "" {
+		ignoredKeys = append(ignoredKeys, "build.isolation")
+	}
+
+	if build.Tags != nil {
+		ignoredKeys = append(ignoredKeys, "build.tags")
+	}
+
+	if build.Extensions != nil {
+		// catchall for any unrecognized key
+		for ext := range build.Extensions {
+			ignoredKeys = append(ignoredKeys, "build."+ext)
+		}
+	}
+
+	if build.Labels != nil {
+		ignoredKeys = append(ignoredKeys, "build.labels")
+	}
+
+	if len(ignoredKeys) == 0 {
+		return nil, nil
```

Comment:
Why do we return `nil` 🤔

Reply:
It made testing more convenient (since `nil` is the empty value for maps) but I found a different way to do it that allows us to remove this.